http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/resources/PlainComplexClientResource.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.ts.http.restclient.reactive.resources;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import io.quarkus.ts.http.restclient.reactive.ResourceAndSubResourcesClient;
+
+@Path("clients/{clientId}/clientResource")
+@Consumes("text/plain")
+@Produces("text/plain")
+public class PlainComplexClientResource {
+    @Inject
+    @RestClient
+    ResourceAndSubResourcesClient resourceAndSubResourcesClient;
+
+    @DELETE
+    @Path("/{id}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String deleteClientResource(@PathParam("clientId") String rootParam,
+            @PathParam("id") String id,
+            @QueryParam("baseUri") String baseUri) throws URISyntaxException {
+        if (!StringUtils.isEmpty(baseUri)) {
+            ResourceAndSubResourcesClient rClient = RestClientBuilder.newBuilder().baseUri(new URI(baseUri))
+                    .build(ResourceAndSubResourcesClient.class);
+            return rClient.clients().get(rootParam).sub().findById(id).remove();
+        }
+
+        return resourceAndSubResourcesClient.clients().get(rootParam).sub().findById(id).remove();
```

Comment:
Lines 41 and 38 are almost identical. Can we put the client into a separate variable and have only one return statement?